### ISSUE TITLE
Rustified enums getting both #[repr(u32)] and #[repr(C)]

### ISSUE THREAD

**Issue by faern** [rust-lang-nursery/rust-bindgen#1224]
### Input C header

```C
enum my_enum {
    VARIANT1,
};
```

### Bindgen Invocation

```
$ bindgen enum.h --rustified-enum my_enum
```

### Actual Results

With `bindgen 0.32.1` I get:
```rust
#[repr(u32)]
#[repr(C)]
#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
pub enum my_enum {
    VARIANT1 = 0,
}
```

rustc warning when compiling:
```
warning[E0566]: conflicting representation hints
 --> enum.rs:3:8
  |
3 | #[repr(u32)]
  |        ^^^
4 | #[repr(C)]
  |        ^
```

### Expected Results

To only have one `#[repr(...)]`:
```rust
#[repr(C)]
#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
pub enum my_enum {
    VARIANT1 = 0,
}
```

**Comment by faern** [comment]
The same problem can be seen in the generated code in issue #1198. But that issue is about something else, so I felt it was worth reporting on this alone.

**Comment by RazrFalcon** [comment]
Stumbled on the same bug just now.

**Comment by emilio** [comment]
Hmm, this is clearly a regression... Shouldn't be hard to fix.

**Comment by emilio** [comment]
Oh, I lie, this was added intentionally in 43c43ba80... Which version of rustc are you using?

**Comment by faern** [comment]
@emilio `rustc 1.25.0-nightly (3f92e8d89 2018-01-14)`

**Comment by emilio** [comment]
Ok, asked in `#rustc` today. Relevant log: https://botbot.me/mozilla/rustc/2018-01-19/?msg=95934948&page=2

TL;DR: the extra repr(C) is only needed for enums with fields, not for plain enums that bindgen generates.

**Comment by emilio** [comment]
Will send a patch with a fix.
